generateLiteloader.py

```diff
@@ -1,93 +1,108 @@
-from meta.liteloaderutil import *
+import os
+from datetime import datetime
+from typing import List, Tuple, Dict, Optional
+
 from meta.common import ensure_component_dir, polymc_path, upstream_path
+from meta.common.liteloader import LITELOADER_COMPONENT, VERSIONS_FILE
+from meta.common.mojang import MINECRAFT_COMPONENT
+from meta.model import MetaVersion, GradleSpecifier, Library, MetaPackage, Dependency
+from meta.model.liteloader import LiteloaderIndex, LiteloaderArtefact
 
 PMC_DIR = polymc_path()
 UPSTREAM_DIR = upstream_path()
 
-ensure_component_dir("com.mumfrey.liteloader")
-
-
-# load the locally cached version list
-def loadLiteloaderJson():
-    with open(UPSTREAM_DIR + "/liteloader/versions.json", 'r', encoding='utf-8') as f:
-        return LiteloaderIndex(json.load(f))
+ensure_component_dir(LITELOADER_COMPONENT)
 
 
-remoteVersionlist = loadLiteloaderJson()
-
-
-def processArtefacts(mcVersion, liteloader, notSnapshots):
-    versions = []
-    lookup = {}
-    latestVersion = None
+def process_artefacts(mc_version: str, artefacts: Dict[str, LiteloaderArtefact], is_snapshot: bool) \
+        -> Tuple[List[MetaVersion], Optional[MetaVersion]]:
+    versions: List[MetaVersion] = []
+    lookup: Dict[str, MetaVersion] = {}
+    latest_version = None
     latest = None
-    for id, artefact in liteloader.items():
-        if id == 'latest':
-            latestVersion = artefact.version
+    for x, artefact in artefacts.items():
```

Comment:
Refactor the `artefact` variables and functions to `artifact` pls?

Reply:
LiteLoader calls it "Artefact" which is why we also use the same names here.

Reply:
Though we could rename all of that to artifact and define `artefact` as an alias in the model

Reply:
Oh it's `artefact` in the jsons so no worries